core/src/banking_stage/unprocessed_transaction_storage.rs

```diff
@@ -1128,24 +1128,28 @@ impl BundleStorage {
         deserialized_bundles: Vec<ImmutableDeserializedBundle>,
         push_back: bool,
     ) -> InsertPacketBundlesSummary {
-        let mut num_bundles_inserted: usize = 0;
-        let mut num_packets_inserted: usize = 0;
-        let mut num_bundles_dropped: usize = 0;
-        let mut num_packets_dropped: usize = 0;
-
-        for bundle in deserialized_bundles {
-            if deque.capacity() == deque.len() {
-                saturating_add_assign!(num_bundles_dropped, 1);
-                saturating_add_assign!(num_packets_dropped, bundle.len());
-            } else {
-                saturating_add_assign!(num_bundles_inserted, 1);
-                saturating_add_assign!(num_packets_inserted, bundle.len());
-                if push_back {
-                    deque.push_back(bundle);
-                } else {
-                    deque.push_front(bundle)
-                }
-            }
+        let deque_bundle_free_space = deque.capacity() - deque.len();
+        let bundles_to_insert_count =
+            std::cmp::min(deque_bundle_free_space, deserialized_bundles.len());
+        let num_bundles_dropped = deserialized_bundles.len() - bundles_to_insert_count;
+        let num_packets_inserted = deserialized_bundles
```

Comment:
kinda curious if we take a non-negligible perf hit going from one loop to now 3 loops ran serially? this code is somehwat of a hot path

Reply:
benchmarked old and new, see main PR description.